pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -929,6 +929,16 @@ func combineTaskRunAndTaskSpecLabels(pr *v1beta1.PipelineRun, pipelineTask *v1be
 			labels[key] = value
 		}
 	}
+
+	// check if a task is part of the finally section, add a label to identify it during the runtime
+	if pr.Status.PipelineSpec != nil {
```

Comment:
If `nil` we just ignore it and go on?

Reply:
This `if` statement is just avoiding `nil` pointer reference and this logic is mainly for the `finally` task. By the time, a `finally` taskRun is getting created, the `status` is guaranteed to have the specification (which should be initialized before the first `taskRun` is created).